FirstApp/Model/User.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  User.swift
+//  FirstApp
+//
+//  Created by Akua Afrane-Okese on 2020/02/27.
+//  Copyright © 2020 Akua Afrane-Okese. All rights reserved.
+//
+
+import Foundation
+import FirebaseAuth
+import Firebase
+
+struct User {
+    public var firstName: String
+    public var lastName: String
+    public var email: String
+    public var password: String
+
+    func addUser(with firstName: String, lastName: String, email: String, and password: String) -> String {
+        var errorMessage = ""
+        Auth.auth().createUser(withEmail: email, password: password) { (result, err) in
+        //Check for errors
+            if err != nil {
+                //There was an error creating the user
+                errorMessage = "Error creating user"
+            } else {
+                //User was created successfully, now store the first name and last name
+                let database = Firestore.firestore()
+                database.collection("users").addDocument(data: ["firstname":
+                firstName, "lastname": lastName,
+                           "uid": result!.user.uid]) { (error) in
+                    //if not nil then there was an error with creating first name and last name
+                    if error != nil {
+                        //Show error message
+                        errorMessage = "Error saving user data"
+                    }
+                }
+            }
+        }
+        return errorMessage
+    }
```

Comment:
I would not put this function in the User Model. The action of adding a user to a data base is of no concern to a user. It should be in some form of manager or other service caller.

Reply:
It's how PJ suggested I create it. I actually do not need a model for the functionality to work but for the purposes of an MVVM structure, PJ made the suggestion.